game.cpp

```diff
@@ -1351,18 +1326,72 @@ void test_game() //!OCLINT tests may be many
     assert(player_copy.get_speed() == p.get_speed());
   }
 
-  /// When a player is killed it stays in the player vector but its state is dead
+  /// When a player is out it stays in the player vector but its state is out
   {
     game g;
 
     auto num_of_players_begin = g.get_v_player().size();
 
-    //kill the first player
     g.kill_player(0);
 
     assert(num_of_players_begin == g.get_v_player().size());
-    assert(is_dead(g.get_player(0)));
+    assert(is_out(g.get_player(0)));
+  }
+
+#ifdef FIX_ISSUE_606
+  {
+    // (626) A player that is out can be revived
+    game g;
+    g.kill_player(0);
+    assert(is_out(g.get_player(0)));
```

Comment:
I am unsure why `is_out` is used over `is_dead`. I assume there is some smart reasoning behind this :-)

Reply:
Yeah I thought "out" is more appropriate since "dead" would imply the player is eliminated. Byt reading through the game logic it doesn't sound so; I think I'm goin to revert back to "dead"